src/type_conversion/jsproxy.c

```diff
@@ -544,8 +638,42 @@ JsProxy_AsJs(PyObject* x)
   return hiwire_incref(js_proxy->js);
 }
 
+int
+JsException_Check(PyObject* x)
+{
+  return PyObject_TypeCheck(x, (PyTypeObject*)Exc_JsException);
+}
+
+int
+JsException_AsJs(PyObject* err)
+{
+  JsExceptionObject* err_obj = (JsExceptionObject*)err;
+  JsProxy* js_error = (JsProxy*)(err_obj->js_error);
+  return hiwire_incref(js_error->js);
+}
+
 int
 JsProxy_init()
 {
-  return (PyType_Ready(&JsProxyType) || PyType_Ready(&JsBoundMethodType));
+  PyExc_BaseException_Type = (PyTypeObject*)PyExc_BaseException;
+  _Exc_JsException.tp_base = (PyTypeObject*)PyExc_Exception;
+
+  PyObject* module;
+  PyObject* exc;
+
+  module = PyImport_ImportModule("pyodide");
+  if (module == NULL) {
+    goto fail;
+  }
+  if (PyObject_SetAttrString(module, "JsException", Exc_JsException)) {
```

Comment:
Could you please add a comment why we need to do this?

Reply:
Yes. (This is in case someone wants to catch `JsException`.)